lib/commands/gesture.js

```diff
@@ -25,8 +39,199 @@ commands.click = async function (el) {
   if (result.status === youiEngineDriverReturnValues.WEBDRIVER_STALE_ELEMENT)
     throw new errors.StaleElementReferenceError();
 
+  return result.value;
+};
+
+commands.tapElXY = async function (el, x, y, longPress) {
+
+  // perform a tap on the given element at the given x,y offset
+  // if longPress is true, the tap becomes a longPress action
+
+  el = unwrapEl(el);
+
+  let commandName = 'tap';
+  if (longPress === true) {
+    commandName = 'longpress';
+  }
+
+  var commandObject = {
+    name: commandName,
+    args: [`${el}`, `${x}`, `${y}`]
+  };
+
+  var commandJSON = JSON.stringify(commandObject);
+  let data = await this.executeSocketCommand(commandJSON);
+
+  let result;
+  try {
+    result = JSON.parse(data);
+  } catch (e) {
+    throw new Error("Bad response from tapElXY");
+  }
+
+  if (result.status === youiEngineDriverReturnValues.WEBDRIVER_UNKNOWN_COMMAND) {
+    throw new errors.UnknownCommandError();
+  } else if (result.status === youiEngineDriverReturnValues.WEBDRIVER_NO_SUCH_ELEMENT) {
+    throw new errors.NoSuchElementError();
+  } else if (result.status === youiEngineDriverReturnValues.WEBDRIVER_STALE_ELEMENT) {
+    throw new errors.StaleElementReferenceError();
+  }
 
   return result.value;
+};
+
+commands.tapXY = async function(x, y, longPress) {
+
+  // perform a tap at the given x,y screen coordinate
+  // if longPress is true, the tap becomes a longPress action
+
+  let commandName = 'tap';
+  if (longPress === true) {
+    commandName = 'longpress';
+  }
 
+  var commandObject = {
+    name: commandName,
+    args: [`${x}`, `${y}`]
+  };
+
+  var commandJSON = JSON.stringify(commandObject);
+  let data = await this.executeSocketCommand(commandJSON);
+
+  let result;
+  try {
+    result = JSON.parse(data);
+  } catch (e) {
+    throw new Error("Bad response from tapXY");
+  }
+
+  // get status returned
+  if (result.status === youiEngineDriverReturnValues.WEBDRIVER_STALE_ELEMENT)
+    throw new errors.StaleElementReferenceError();
+
+  return result.value;
 };
+
+commands.tap = async function (gestures, longPress) {
+
+  // parse the given gestures array to call the appropriate tap method
+  // if longPress is true, the tap is a long press action
+
+  let elementId = gestures[0].options.element;
+  let x = gestures[0].options.x;
+  let y = gestures[0].options.y;
+
+  if (elementId) {
+
+    // we are either tapping on the default location of the element
+    // top left has been specified of the element has been specified
+
+    if (x && y) {
+      await this.tapElXY(elementId, x, y, longPress);
+    } else {
+      // tapping with an element parameter is the same as click on a WebElement so they can
+      // filter through the same call stack
```

Comment:
`so they can fiilter through the same call stack` This doesn't add any information.